crates/utils/re_byte_size/src/arrow_sizes.rs

```diff
@@ -15,3 +19,17 @@ impl SizeBytes for ArrayRef {
         self.get_array_memory_size() as u64
     }
 }
+
+impl<T: ArrowPrimitiveType> SizeBytes for PrimitiveArray<T> {
+    #[inline]
+    fn heap_size_bytes(&self) -> u64 {
+        Array::get_array_memory_size(self) as u64
+    }
+}
+
+impl<T: ArrowNativeType> SizeBytes for ScalarBuffer<T> {
+    #[inline]
+    fn heap_size_bytes(&self) -> u64 {
+        self.inner().capacity() as _
```

Comment:
We're a bit inconsistent wether or not we measure used memory (`.len()`) or allocated memory (`.capcity()`). I think we should always do the latter. WDYT @teh-cmc ?